blocks/components/editable/tinymce.js

```diff
@@ -19,12 +22,42 @@ export default class TinyMCE extends wp.element.Component {
 		}
 	}
 
-	shouldComponentUpdate() {
+	componentDidMount() {
+		this.initialize();
+	}
+
+	shouldComponentUpdate( nextProps ) {
 		// We must prevent rerenders because TinyMCE will modify the DOM, thus
 		// breaking React's ability to reconcile changes.
 		//
 		// See: https://github.com/facebook/react/issues/6802
-		return false;
+		return nextProps.tagName !== this.props.tagName;
+	}
+
+	componentWillUpdate( nextProps ) {
+		if ( ! this.editor ) {
+			return;
+		}
+
+		if ( nextProps.tagName !== this.props.tagName ) {
+			this.editor.destroy();
```

Comment:
Is this how React is able to reconcile? That we're unmounting the node altogether and React recreates it in the next pass?